static/app/components/onboarding/gettingStartedDoc/layout.tsx

```diff
@@ -38,20 +46,43 @@ export type LayoutProps = {
   platformKey?: PlatformKey;
 };
 
+function getDefaultSelectedProducts({
+  includeProfiling,
+  hideSessionReplay,
+}: {
+  hideSessionReplay?: boolean;
+  includeProfiling?: boolean;
+}) {
+  const defaultSelectedProducts = [ProductSolution.PERFORMANCE_MONITORING];
+
+  if (includeProfiling) {
+    defaultSelectedProducts.push(ProductSolution.PROFILING);
+  }
+
+  if (!hideSessionReplay) {
+    defaultSelectedProducts.push(ProductSolution.SESSION_REPLAY);
+  }
+
+  return defaultSelectedProducts;
+}
+
 export function Layout({
   steps,
   platformKey,
   nextSteps = [],
   newOrg,
   introduction,
+  includeProfiling,
+  hideSessionReplay,
 }: LayoutProps) {
   const organization = useOrganization();
   const {isSelfHosted} = useLegacyStore(ConfigStore);
 
   const isJavaScriptPlatform =
     platformKey === 'javascript' || !!platformKey?.match('^javascript-([A-Za-z]+)$');
 
-  const displayProductSelection = !isSelfHosted && isJavaScriptPlatform;
+  const displayProductSelection =
+    !isSelfHosted && (isJavaScriptPlatform || includeProfiling);
```

Comment:
These one-off booleans (isJavaScriptPlatform, includeProfiling, hideSessionReplay) are no longer sustainable.
How about an object where we have SDK names as keys and an array of available products as values? If the SDK is not present (or the array is empty), that platform does not have a product selector.

Reply:
yes, I noticed the same and wanted to change that in the same PR, but in the end I decided to work on it in a separated one. your idea is good. will implement that. thanks 🙏

Reply:
Let's please not ship like this and do it right in this PR 🙏

Reply:
as we talked, will add this change in a separate PR and merge before this 😉